katalog/fluentd/kubernetes.conf

```diff
@@ -167,11 +169,23 @@
   pos_file /var/log/kube-apiserver-audit.log.pos
   tag kube-apiserver-audit
   <parse>
-    @type json
+    @type multiline
+    format_firstline /^\S+\s+AUDIT:/
+    # Fields must be explicitly captured by name to be parsed into the record.
+    # Fields may not always be present, and order may change, so this just looks
+    # for a list of key="\"quoted\" value" pairs separated by spaces.
+    # Unknown fields are ignored.
+    # Note: We can't separate query/response lines as format1/format2 because
+    #       they don't always come one after the other for a given query.
+    format1 /^(?<time>\S+) AUDIT:(?: (?:id="(?<id>(?:[^"\\]|\\.)*)"|ip="(?<ip>(?:[^"\\]|\\.)*)"|method="(?<method>(?:[^"\\]|\\.)*)"|user="(?<user>(?:[^"\\]|\\.)*)"|groups="(?<groups>(?:[^"\\]|\\.)*)"|as="(?<as>(?:[^"\\]|\\.)*)"|asgroups="(?<asgroups>(?:[^"\\]|\\.)*)"|namespace="(?<namespace>(?:[^"\\]|\\.)*)"|uri="(?<uri>(?:[^"\\]|\\.)*)"|response="(?<response>(?:[^"\\]|\\.)*)"|\w+="(?:[^"\\]|\\.)*"))*/
+    time_format %Y-%m-%dT%T.%L%Z
```

Comment:
Any reason for this?

Reply:
Just copied the upstream configuration: https://github.com/fluent/fluentd-kubernetes-daemonset/blob/master/docker-image/v1.10/debian-elasticsearch6/conf/kubernetes.conf#L167-L189

Also the Kubernetes official site purpose another fluentd configuration 🤦 : https://v1-16.docs.kubernetes.io/docs/tasks/debug-application-cluster/audit/#log-collector-examples

I am double checking our configuration:
on-premise installer: https://github.com/sighupio/fury-kubernetes-on-premise/blob/master/roles/kube-control-plane/templates/kubeadm.yml.j2#L27
aws installer: https://github.com/sighupio/fury-kubernetes-aws/blob/master/roles/kube-control-plane/templates/kubeadm.yml.j2#L32-L36

At this point, I am not sure if it is currently working....

So we have a couple of options:
1. Rollback this small change
2. Go forward with upstream

IMHO: Going forward with upstream is the right way to go

Reply:
I agree that we should follow upstream, can we test that the parsing is working with our audit configuration?

Reply:
Witch which Kubernetes version should i test it? What audit configuration should i choose? The AWS one or the onpremise?

Reply:
The files which contain the audit policy are:
- https://github.com/sighupio/fury-kubernetes-on-premise/blob/master/roles/kube-control-plane/files/audit.yml
- https://github.com/sighupio/fury-kubernetes-aws/blob/master/roles/kube-control-plane/files/audit.yml
and they're are equal between aws and on-premise.

The files you linked unfortunately use two different apiVersion for kubeadm configuration but the end result is the same.

Reply:
Tested and is NOT working with the upstream configuration. I rollback (to json) it and now is working again!

Reply:
Thanks a lot @lnovara!